test/next_ls_test.exs

```diff
@@ -235,6 +233,9 @@ defmodule NextLSTest do
       assert_result 2, nil
     end
 
+    # dependency should not show up in workspace symbols
+    @tag :with_dependency
```

Comment:
I think it is more flexible to make the tag take the dependencies itself

```elixir
@tag dependencies: """
{:jason, "~> 1.4"},
{:foo, "~> 2.0"}
"""
```

then in the mix_exs function, you just interpolate that into the string

I would also unpack the new `write_project` function, as I sense this going to be a brittle function that will churn often.